src/uzfs_prop.c

```diff
@@ -0,0 +1,41 @@
+#include <zrepl_mgmt.h>
+#include <uzfs_mgmt.h>
+
+/*
+ * This file defines function to set property of uzfs zvol
+ */
+int
+uzfs_zfs_set_prop(const char *name, zprop_source_t source, nvlist_t *list)
+{
+	zvol_info_t *zinfo;
+	int error;
+
+	/* fetch zinfo for given volume */
+	zinfo = uzfs_zinfo_lookup(name);
+	if (zinfo == NULL) {
+		LOG_ERR("Invalid volume %s", name);
+		return (EINVAL);
+	}
+
+	(void) pthread_mutex_lock(&zinfo->zinfo_mutex);
+	error = update_zvol_property(zinfo->main_zv, list);
+	if (error) {
+		LOG_ERR("Property updation failed(%d)", error);
+		goto end;
+	}
+
+	if (error == 0 && zinfo->clone_zv != NULL) {
```

Comment:
lets just set readonly property at zinfo and main_zv.. this way - we have control of this at one place only in each layer.
And, setting on clone_zv is not required if we remove the check on clone_zv in uzfs_submit_writes.

Reply:
done